pkg/apis/core/validation/shoot.go

```diff
@@ -266,6 +267,9 @@ func ValidateShootSpecUpdate(newSpec, oldSpec *core.ShootSpec, newObjectMeta met
 
 	if newObjectMeta.DeletionTimestamp != nil && !apiequality.Semantic.DeepEqual(newSpec, oldSpec) {
 		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newSpec, oldSpec, fldPath)...)
+		if diff := deep.Equal(newSpec, oldSpec); diff != nil {
+			allErrs = append(allErrs, field.Forbidden(fldPath, strings.Join(diff, ",")))
+		}
```

Comment:
```suggestion
	if newObjectMeta.DeletionTimestamp != nil && !apiequality.Semantic.DeepEqual(newSpec, oldSpec) {
		errorList := apivalidation.ValidateImmutableField(newSpec, oldSpec, fldPath)
		if diff := deep.Equal(newSpec, oldSpec); diff != nil {
			errorList = field.ErrorList{field.Forbidden(fldPath, strings.Join(diff, ","))}
		}
		return errorList
	}
```
This way we can reduce duplication. The first error message doesn't help anyway.